TrashPath/App.js

```diff
@@ -0,0 +1,228 @@
+  const [location, setLocation] = useState(null);
+  const [region, setRegion] = useState(null);
+  const [errorMsg, setErrorMsg] = useState(null);
+  const [appIsReady, setAppIsReady] = useState(false);
+  const [search, setSearch] = useState();
+
+  const [open, setOpen] = useState(false);
+  const [pictureData, setPictureData] = useState(false);
+
+  const cameraRef = useRef(null);
+
+  const onLocationChange = (newLocation) => {
+    if (newLocation?.coords) {
+      setLocation(newLocation);
+      setRegion({
+        latitude: newLocation.coords.latitude,
+        longitude: newLocation.coords.longitude,
+        latitudeDelta: 0.005,
+        longitudeDelta: 0.005,
+      });
+    }
+  };
+
+  const onRegionChange = (newRegion) => {
+    setRegion(newRegion);
+  };
+
+  const updateSearch = (newSearch) => {
+    setSearch(newSearch);
+  };
+
+  const searchforMaterial = (e) => {
+    console.log(search);
+  };
+
+  useEffect(() => {
+    let watcher;
+    async function init() {
+      let { status } = await Location.requestForegroundPermissionsAsync();
+
+      if (status !== "granted") {
+        setErrorMsg("Permission to access location was denied");
+        return;
+      }
+
+      watcher = await Location.watchPositionAsync(
+        { accuracy: 5, distanceInterval: 5 },
+        onLocationChange
+      );
+      setAppIsReady(true);
+    }
+
+    init();
+
+    // cleanup
+    return () => watcher?.remove();
+  }, []);
+
+  useEffect(() => {
+    async () => {
+      let { status } = await Camera.requestPermissionsAsync();
+
+      if (status !== "granted") {
+        setErrorMsg("Permission to access camera was denied");
+        return;
+      }
+    };
+  });
+
+  const openCameraModal = () => {
+    setOpen(true);
+  };
+
+  const handlePictureTaken = (data) => {
+    // TODO
+
+    setPictureData(data);
+
+    setOpen(false);
+  };
+
+  const handleSubmitGeotag = async (data) => {
+    console.log(data);
+
+    // TODO: send to database
+  };
+
+  return (
+    <View style={styles.container}>
+      {region && (
+        <MapView
+          region={region}
+          //onRegionChange={onRegionChange}
+          style={styles.map}
+        >
+          <Marker
+            coordinate={{
+              latitude: location.coords.latitude,
+              longitude: location.coords.longitude,
+            }}
+            image={require("./assets/currentlocation.png")}
+          />
+        </MapView>
+      )}
+
+      <GeotagSubmit pictureData={pictureData} onSubmit={handleSubmitGeotag} />
+
+      <CameraModal open={open} onPictureTaken={handlePictureTaken} />
+
+      <View style={styles.toolbar}>
+        <TouchableOpacity
+          onPress={openCameraModal}
+          style={styles.locationButton}
+        >
+          <Image
+            style={styles.imagestyle}
+            source={require("./assets/addtag.png")}
+          />
+        </TouchableOpacity>
+      </View>
+
+      {/* <Camera
```

Comment:
Is this commented code still needed?

Reply:
Until the geotag submission is done I'm keeping that around for debugging